A Debian-specific patch: the Asterisk datadir is under /usr/share.
This patch still needs improving. 

Better yet: get the settings from Asterisk's build-time settings (?)
--- a/Makefile
+++ b/Makefile
@@ -47,6 +47,9 @@ OTHER_SUBDIR_CFLAGS=$(ASTERISK_INCLUDE)
 # endif
 #endif
 
+# On Debian:
+ASTDATADIR=/usr/share/asterisk
+
 ifneq ($(wildcard makeopts),)
   include makeopts
 endif
